### PR TITLE
auto update rates

### PR DESCRIPTION
**Pull request by malygina**

### CODE REVIEW

File: src/main/java/com/currencybaskets/tasks/UpdateLatestRateScheduledTask.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	private RateRepository rateRepository;
+
+	//Once per day
+	@Scheduled(fixedRate = 24 * 60 * 60 * 60 * 1000L)
+	public void updateLatestRate() {
+		CurrencyRate currencyRate = fixer.getLatestRate();
+		if (!currencyRate.isSuccess()) {
+			throw new ServiceUnavailableException("Fixer service is not available");
+		}
+		LOG.info("Currency rate will be updated");
```
Comment (EgorGavrilov) [inline src/main/java/com/currencybaskets/tasks/UpdateLatestRateScheduledTask.java]: Use lombock SLf4J

File: src/main/java/com/currencybaskets/tasks/UpdateLatestRateScheduledTask.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Component
+public class UpdateLatestRateScheduledTask {
+
+	private static final Logger LOG = LoggerFactory.getLogger(UpdateLatestRateScheduledTask.class);
```
Comment (egorvlgavr) [inline src/main/java/com/currencybaskets/tasks/UpdateLatestRateScheduledTask.java]: Please use Lombok as in this class https://github.com/egorvlgavr/currency-baskets/blob/master/src/main/java/com/currencybaskets/services/AccountService.java#L20

File: src/test/java/com/currencybaskets/services/CurrencyFixerClientTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	@Test
+	public void whenCallingGetLatestRate_thenClientReturnsErrorResponse() throws Exception
+	{
+		JSONObject resp = new JSONObject();
+		JSONObject errorObj = new JSONObject();
+		errorObj.put("code", 104);
+		errorObj.put("info", "Our monthly API request volume has been reached. Please upgrade your plan.");
+		resp.put("success", false);
+		resp.put("error", errorObj);
+		this.server.expect(requestTo("http://data.fixer.io/api/latest?access_key=b0e12ace307602c973391a39f72967ce&symbols=RUB,USD,EUR"))
```
Comment (egorvlgavr) [inline src/test/java/com/currencybaskets/services/CurrencyFixerClientTest.java]: Seems like incorrect key here

File: src/main/java/com/currencybaskets/tasks/UpdateLatestRateScheduledTask.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	public void updateLatestRate() {
+		CurrencyRate currencyRate = fixer.getLatestRate();
+		if (!currencyRate.isSuccess()) {
+			throw new ServiceUnavailableException("Fixer service is not available");
+		}
+		LOG.info("Currency rate will be updated");
+		LatestRate rates = currencyRate.getLatestRate();
+		Long previousRateEurId = rateRepository.findLatestRateIdByCurrencyName("EUR");
+		Long previousRateUsdId = rateRepository.findLatestRateIdByCurrencyName("USD");
+		List<RateUpdate> updates = Stream.of(new RateUpdate(previousRateEurId, rates.getEurInRub()),
```
Comment (egorvlgavr) [inline src/main/java/com/currencybaskets/tasks/UpdateLatestRateScheduledTask.java]: Please log id of rate that will be updated.
Create list with Collections.asList(...)

**Review by egorvlgavr — CHANGES_REQUESTED** [review]

**Review by egorvlgavr — APPROVED** [review]

**Pull request merged**